crates/uv-resolver/src/lock/mod.rs

```diff
@@ -2837,6 +2837,35 @@ struct PackageMetadata {
     dependency_groups: BTreeMap<GroupName, BTreeSet<Requirement>>,
 }
 
+impl PackageMetadata {
+    fn unwire(self, requires_python: &RequiresPython) -> Result<PackageMetadata, LockError> {
+        // We need to complexify these markers so things like
+        // `requires_python < '0'` get normalized to False
+        let unwire_requirements =
+            |requirements: BTreeSet<Requirement>| -> Result<BTreeSet<Requirement>, LockError> {
+                requirements
+                    .into_iter()
+                    .map(|mut requirement| {
+                        let complexified_marker =
+                            requires_python.complexify_markers(requirement.marker);
+                        requirement.marker = complexified_marker;
+                        Ok(requirement)
```

Comment:
All the Results here are actually pointless, but I only realized this at the very end and I'm calling it a night. I should remove them (DependencyWire::unwire is fallible because it resolves packageids, not because it complexifies marker).